src/transformers/models/llama/modeling_llama.py

```diff
@@ -99,8 +99,8 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None):
         freqs = torch.einsum("i,j->ij", t, self.inv_freq)
         # Different from paper, but it uses a different permutation in order to obtain the same calculation
         emb = torch.cat((freqs, freqs), dim=-1)
-        self.cos_cached = emb.cos()[None, None, :, :]
-        self.sin_cached = emb.sin()[None, None, :, :]
+        self.register_buffer("cos_cached", emb.cos()[None, None, :, :], persistent=False)
+        self.register_buffer("sin_cached", emb.sin()[None, None, :, :], persistent=False)
```

Comment:
```suggestion
        self.register_buffer("cos_cached", emb.cos()[None, None, :, :])
        self.register_buffer("sin_cached", emb.sin()[None, None, :, :])
```

Reply:
Hi, I set persistent=False in this PR because `cos_cached` and `sin_cached` are not included in the model's state_dict of the original checkpoint. Setting persistent=True will induce missing key warnings when loading the llama model with from_pretrained(). 
But if this breaks the process of model loading on the meta device, please feel free to correct them.

Reply:
if `persistent=True` is aboslutely needed to make the meta device loading work, you could add `r".*.cos_cached"` in `_keys_to_ignore_on_missing` and `_keys_to_ignore_on_unexpected` (same for `sin_cached`)

Reply:
I've done some testing and this change isn't necessary: those buffers are not affected by `with init_empty_weights():` and the problem was somewhere else.
Consider this suggestion invalid.
Thank you!